File: ew_monero/ios/Classes/EwMoneroPlugin.m
#import "EwMoneroPlugin.h"
#import <ew_monero/ew_monero-Swift.h>

@implementation EwMoneroPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  [SwiftEwMoneroPlugin registerWithRegistrar:registrar];
}
@end

File: ew_monero/macos/Classes/EwMoneroPlugin.swift
import Cocoa
import FlutterMacOS

public class EwMoneroPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "ew_monero", binaryMessenger: registrar.messenger)
    let instance = EwMoneroPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
